### ISSUE TITLE
Multisignature support

### ISSUE THREAD

**Issue by johndoe123445** [emesik/monero-python#45]
Hello,

Thanks for the great work you’ve done there.

As you may know, monero v0.13.0.4 has finally stabilized M-N multisig of monero.

The JSON RPC has `create_wallet, prepare_multisig, make_multisig, finalize_multisig` to create a multisignature wallet.

Can you implement that feature to this wonderful library? I would like to help, but the creation is too complicated for the api to expose the methods clean.

Thanks.

**Comment by johndoe123445** [comment]
https://blog.keys.casa/a-monero-multisig-users-guide/

https://www.getmonero.org/resources/developer-guides/wallet-rpc.html#make_multisig

**Comment by emesik** [comment]
Definitely, multisig has been a part of the plan from the very beginning. As you see, the task is not trivial though. At the moment I can't give any time estimate when it could be done.

MMS also looks to be interesting feature to have implemented, as an add-on: https://rbrunner7.github.io/mms_full_manual

**Comment by johndoe123445** [comment]
How about just a just simple wrapper for these?

create_wallet
prepare_multisig
make_multisig
finalize_multisig

**Comment by emesik** [comment]
Not so easy. The `Wallet` class is designed to run a single wallet only. We would need a `WalletManager` or some other higher-level handler for `create_wallet`. You may look for something similar [in this project](https://github.com/emesik/monerowalletpool). Be aware, however, that code is highly experimental and crashes/hangs on IPC quirks very often.

Also, signing of transactions must be implemented, and we should distinguish signed and unsigned txs. I touched the topic in [unsigned-tx](https://github.com/monero-ecosystem/monero-python/tree/unsigned-tx) branch but had no more time to work on it.

Certainly, it is an important feature but we need it to be designed well. If you, or anyone else, find the code in `monerowalletpool` useful here, I'm willing to release necessary parts under BSD license too.

**Comment by philogy** [comment]
What's the development status of the multisig feature?

**Comment by emesik** [comment]
Not progressing. I'd love to have time to spend on it.

**Comment by h0ay** [comment]
is there any progress on monero multi sig support ?

**Comment by emesik** [comment]
None. I'm going to release 1.0 soon as there are no outstanding bug reports. Then I'll focus on new features but recently I don't have much time to spend on it.

**Comment by h0ay** [comment]
Lemme know if i can help you in anyway ,donations ,contributions

**Comment by emesik** [comment]
Actually I'm thinking about applying for funding. However, first I need to carve out time to read docs and estimate the effort.

**Comment by h0ay** [comment]
That's cool 
Message me when you need the hep

**Comment by h0ay** [comment]
any discord or instant messaging contact ? 
I could help you to get it started
